<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>MainWindow</class>
    <widget class="QMainWindow" name="MainWindow">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>800</width>
                <height>600</height>
            </rect>
        </property>
        <property name="windowTitle">
            <string>Packet Sniffer</string>
        </property>
        <widget class="QWidget" name="centralwidget">
            <layout class="QVBoxLayout" name="verticalLayout">
                <item>
                    <widget class="QFrame" name="captureOptionsFrame">
                        <layout class="QHBoxLayout" name="horizontalLayout">
                            <item>
                                <widget class="QLabel" name="protocolLabel">
                                    <property name="text">
                                        <string>Protocol:</string>
                                    </property>
                                </widget>
                            </item>
                            <item>
                                <widget class="QComboBox" name="protocolComboBox">
                                    <item>
                                        <property name="text">
                                            <string>All</string>
                                        </property>
                                    </item>
                                    <item>
                                        <property name="text">
                                            <string>TCP</string>
                                        </property>
                                    </item>
                                    <item>
                                        <property name="text">
                                            <string>UDP</string>
                                        </property>
                                    </item>
                                    <item>
                                        <property name="text">
                                            <string>ICMP</string>
                                        </property>
                                    </item>
                                </widget>
                            </item>
                            <item>
                                <widget class="QLabel" name="interfaceLabel">
                                    <property name="text">
                                        <string>Interface:</string>
                                    </property>
                                </widget>
                            </item>
                            <item>
                                <widget class="QComboBox" name="interfaceComboBox" />
                            </item>
                            <item>
                                <widget class="QPushButton" name="refreshPortsButton">
                                    <property name="text">
                                        <string>Refresh Ports</string>
                                    </property>
                                </widget>
                            </item>
                            <item>
                                <widget class="QPushButton" name="startButton">
                                    <property name="text">
                                        <string>Start Capture</string>
                                    </property>
                                </widget>
                            </item>
                        </layout>
                    </widget>
                </item>
                <item>
                    <widget class="QSplitter" name="splitter">
                        <property name="orientation">
                            <enum>Qt::Vertical</enum>
                        </property>
                        <widget class="QTableWidget" name="packetTable">
                            <column>
                                <property name="text">
                                    <string>No.</string>
                                </property>
                            </column>
                            <column>
                                <property name="text">
                                    <string>Timestamp</string>
                                </property>
                            </column>
                            <column>
                                <property name="text">
                                    <string>Source</string>
                                </property>
                            </column>
                            <column>
                                <property name="text">
                                    <string>Destination</string>
                                </property>
                            </column>
                            <column>
                                <property name="text">
                                    <string>Protocol</string>
                                </property>
                            </column>
                            <column>
                                <property name="text">
                                    <string>Length</string>
                                </property>
                            </column>
                        </widget>
                        <widget class="QTabWidget" name="detailsTabWidget">
                            <property name="sizePolicy">
                                <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                                    <horstretch>0</horstretch>
                                    <verstretch>0</verstretch>
                                </sizepolicy>
                            </property>
                            <widget class="QWidget" name="portsTab">
                                <attribute name="title">
                                    <string>Open Ports</string>
                                </attribute>
                                <layout class="QVBoxLayout" name="verticalLayout_2">
                                    <item>
                                        <widget class="QTableWidget" name="portsTable">
                                            <column>
                                                <property name="text">
                                                    <string>Port</string>
                                                </property>
                                            </column>
                                            <column>
                                                <property name="text">
                                                    <string>Protocol</string>
                                                </property>
                                            </column>
                                            <column>
                                                <property name="text">
                                                    <string>State</string>
                                                </property>
                                            </column>
                                            <column>
                                                <property name="text">
                                                    <string>Service</string>
                                                </property>
                                            </column>
                                        </widget>
                                    </item>
                                </layout>
                            </widget>
                            <widget class="QWidget" name="packetDetailsTab">
                                <attribute name="title">
                                    <string>Packet Details</string>
                                </attribute>
                                <layout class="QVBoxLayout" name="verticalLayout_3">
                                    <item>
                                        <widget class="QTextEdit" name="packetDetailsText">
                                            <property name="readOnly">
                                                <bool>true</bool>
                                            </property>
                                            <property name="font">
                                                <font>
                                                    <family>Monospace</family>
                                                    <pointsize>10</pointsize>
                                                </font>
                                            </property>
                                            <property name="lineWrapMode">
                                                <enum>QTextEdit::NoWrap</enum>
                                            </property>
                                        </widget>
                                    </item>
                                </layout>
                            </widget>
                        </widget>
                    </widget>
                </item>
            </layout>
        </widget>
        <widget class="QStatusBar" name="statusbar" />
        <widget class="QMenuBar" name="menuBar">
            <property name="geometry">
                <rect>
                    <x>0</x>
                    <y>0</y>
                    <width>800</width>
                    <height>20</height>
                </rect>
            </property>
            <widget class="QMenu" name="menuFile">
                <property name="title">
                    <string>File</string>
                </property>
                <addaction name="actionOpen" />
                <addaction name="actionSave" />
                <addaction name="actionSave_As" />
            </widget>
            <widget class="QMenu" name="menuCapture">
                <property name="title">
                    <string>Capture</string>
                </property>
            </widget>
            <widget class="QMenu" name="menuHelp">
                <property name="title">
                    <string>Help</string>
                </property>
            </widget>
            <widget class="QMenu" name="menuView">
                <property name="title">
                    <string>View</string>
                </property>
            </widget>
            <addaction name="menuFile" />
            <addaction name="menuCapture" />
            <addaction name="menuView" />
            <addaction name="menuHelp" />
        </widget>
        <action name="actionOpen">
            <property name="text">
                <string>Open</string>
            </property>
        </action>
        <action name="actionSave">
            <property name="text">
                <string>Save</string>
            </property>
        </action>
        <action name="actionSave_As">
            <property name="text">
                <string>Save As</string>
            </property>
        </action>
    </widget>
    <resources />
    <connections />
</ui>